testnet/stacks-node/src/tests/epoch_205.rs

```diff
@@ -913,3 +916,292 @@ fn test_cost_limit_switch_version205() {
+
+    eprintln!("Chain bootstrapped...");
+
+    let mut run_loop = neon::RunLoop::new(conf);
+    let blocks_processed = run_loop.get_blocks_processed_arc();
+    let microblocks_processed = run_loop.get_microblocks_processed_arc();
+
+    let channel = run_loop.get_coordinator_channel().unwrap();
+
+    thread::spawn(move || run_loop.start(None, 0));
+
+    // give the run loop some time to start up!
+    wait_for_runloop(&blocks_processed);
+
+    // first block wakes up the run loop
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // first block will hold our VRF registration
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // second block will be the first mined Stacks block
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // let's query the miner's account nonce:
+    let account = get_account(&http_origin, &miner_account);
+    assert_eq!(account.nonce, 1);
+    assert_eq!(account.balance, 0);
+
+    for spender_addr in spender_addrs.iter() {
+        let account = get_account(&http_origin, &spender_addr);
+        assert_eq!(account.nonce, 0);
+        assert_eq!(account.balance, 10492300000);
+    }
+
+    let mut ctr = 0;
+    while ctr < txs.len() {
+        submit_tx(&http_origin, &txs[ctr]);
+        if !wait_for_microblocks(&microblocks_processed, 180) {
+            // we time out if we *can't* mine any more microblocks
+            break;
+        }
+        ctr += 1;
+    }
+
+    // only one fit
+    assert_eq!(ctr, 1);
+    sleep_ms(5_000);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    eprintln!("First confirmed microblock stream!");
+
+    microblocks_processed.store(0, Ordering::SeqCst);
+
+    // send the rest of the transactions
+    while ctr < txs.len() {
+        submit_tx(&http_origin, &txs[ctr]);
+        ctr += 1;
+    }
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    eprintln!("Second confirmed microblock stream!");
+
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    // confirm it
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    eprintln!("expect epoch transition");
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    eprintln!("expect microblock stream");
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    wait_for_microblocks(&microblocks_processed, 180);
+
+    // this test can sometimes miss a mine block event.
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let blocks = test_observer::get_blocks();
+    assert!(blocks.len() >= 5, "Should have produced at least 5 blocks");
+
+    let mut max_big_txs_per_microblock_20 = 0;
```

Comment:
seems like you are using the word "large" in the code so maybe `s/big/large/` for these variables